### PR TITLE
Allure integration

### PR DESCRIPTION
**Pull request by Malinskiy**
* integration with allure-kotlin
* upgrade android samples dependencies
* handle exceptions from vendor modules
* map exceptions from vendor-android adam
* handle `am instrument` abrupt finish in adam

### CODE REVIEW

File: vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/configuration/AllureConfiguration.kt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.malinskiy.marathon.android.configuration
+
+val DEFAULT_ALLURE_CONFIGURATION = AllureConfiguration(
+    allureAndroidSupport = false
+)
+
+data class AllureConfiguration(
+    val allureAndroidSupport: Boolean,
```
Comment (Malinskiy) [inline vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/configuration/AllureConfiguration.kt]: Maybe this should be `enabled`?

File: vendor/vendor-android/ddmlib/src/main/kotlin/com/malinskiy/marathon/android/ddmlib/DdmlibAndroidDevice.kt [hunk 2]
```diff
@@ -107,6 +108,32 @@ class DdmlibAndroidDevice(
         }
     }
 
+    override suspend fun pullFolder(remoteFolderPath: String, localFolderPath: String) {
```
Comment (Malinskiy) [inline vendor/vendor-android/ddmlib/src/main/kotlin/com/malinskiy/marathon/android/ddmlib/DdmlibAndroidDevice.kt:111]: How do we make sure that people know this doesn't support following symlinks

File: vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/configuration/AllureConfiguration.kt (hunk 1 above)
Comment (tagantroy) [reply vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/configuration/AllureConfiguration.kt]: I would prefer enabled instead.

**Review by tagantroy — APPROVED** [review]

File: vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/configuration/AllureConfiguration.kt (hunk 1 above)
Comment (Malinskiy) [reply vendor/vendor-android/base/src/main/kotlin/com/malinskiy/marathon/android/configuration/AllureConfiguration.kt]: Let me change this then

File: sample/android-app/Marathonfile [hunk 3]
```diff
@@ -15,4 +15,6 @@ vendorConfiguration:
   testApplicationPmClear: true
   vendor: ADAM
   waitForDevicesTimeoutMillis: 60000
+  allureConfiguration:
+    allureAndroidSupport: true
```
Comment (tagantroy) [inline sample/android-app/Marathonfile]: enabled?

Comment (Malinskiy) [reply sample/android-app/Marathonfile]: done

**Pull request merged**